core/src/main/java/hudson/logging/LogRecorder.java

```diff
@@ -85,16 +87,60 @@
         return ts;
     }
 
+    @Restricted(NoExternalUse.class)
+    @VisibleForTesting
+    public static Set<String> getAutoCompletionCandidates(List<String> loggerNamesList) {
```

Comment:
As the order is kept, I think it could be nice to either explicit that in the name of the method or return a `SortedSet`